### DISCUSSION TITLE
Unable to link libPhysXGpu_64.so on linux

### DISCUSSION THREAD

**Discussion by Spatchler** [NVIDIA-Omniverse/PhysX#314] [Q&A] [answered]
I've build PhysX 5.4.1, all the snippets work fine. But I have been unable to link PhysX into my own C++ project. I have been using premake5.

My project builds just fine but only with cuda and PhysXGpu_64 linked. However when I run the binary I get this error:"error while loading shared libraries: libPhysXGpu_64.so: cannot open shared object file: No such file or directory"

My premake config: https://pastebin.com/00GgeqLV

**Comment by vreutskyy** [comment]
Hi @Spatchler 
As I understand, if all the snippets work, libPhysXGpu_64.so exists. It's just your app that can't find it. Did you try to add the path to PhysX's libraries to LD_LIBRARY_PATH environment variable, or just copy them to where your app is located?

**Reply by Spatchler** [reply]
I haven't set an LD_LIBRARY_PATH environment variable. The library files are stored in the install folder when I built the library and I just add the lib directories in my premake config.

**Reply by Spatchler** [reply, marked as answer]
Ok, I have never encountered LD_LIBRARY_PATH this before when linking shared libraries, probably because I install them with my package manager. I ended up copying the libPhysXGpu_64.so into /usr/lib to fix the issue. Thanks for the reply @vreutskyy

**Comment by foxy993** [comment]
Hi @Spatchler,

I'm currently working on a project using PyPhysX (a Python wrapper around PhysX 4.1.1), and I'm running into the similar issue you described `Failed to load libPhysXGpu_64.so`. It seems that this binary is not built during installation and is not included in the official PhysX repository.  Unfortunately, I wasn't able to locate `libPhysXGpu_64.so` anywhere online.
You mentioned you had this file. Would you be willing to share it (privately is totally fine)? It's only for academic purposes and testing GPU simulations. I'd be extremely grateful :)

**Comment by vreutskyy** [comment]
I believe, before PhysXGpu_64 library source was included in PhysX distribution (PhysX 5.6), generate_projects.bat(.sh) was downloading the correct version of it according to the version and platform you're building for. Could you try to run it and see if it helps?
